src/common/legacy/src/graph_transformer.cpp

```diff
@@ -474,6 +474,22 @@ void ConstTransformer::fullTrim() {
     std::lock_guard<std::mutex> lock(lockFullTrim);
     auto sortedLayers = details::CNNSubnetSortTopologically({inputs, outputs});
     auto constMapLayers = getConstLayers(sortedLayers);
+
+    for (auto& layer : sortedLayers) {
+        [&] {
+            if (layer->type == "Const") {
+                for (auto& out : outputs) {
+                    for (auto& out_const : layer->outData) {
+                        if (out_const == out) {
+                            constMapLayers.erase(layer->name);
+                            return;
+                        }
+                    }
+                }
+            }
+        }();
+    }
+
     auto constData = getConstData(constMapLayers, sortedLayers);
```

Comment:
can you please add test to cover this case?

Reply:
Okay, sure. Thanks!

Reply:
Done, thanks!